graylog2-web-interface/src/views/components/visualizations/hooks/useWidgetUnits.ts

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright (C) 2020 Graylog, Inc.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the Server Side Public License, version 1,
+ * as published by MongoDB, Inc.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * Server Side Public License for more details.
+ *
+ * You should have received a copy of the Server Side Public License
+ * along with this program. If not, see
+ * <http://www.mongodb.com/licensing/server-side-public-license>.
+ */
+import { useMemo } from 'react';
+import flatten from 'lodash/flatten';
+import compact from 'lodash/compact';
+import uniq from 'lodash/uniq';
+
+import useFieldTypesUnits from 'views/hooks/useFieldTypesUnits';
+import type AggregationWidgetConfig from 'views/logic/aggregationbuilder/AggregationWidgetConfig';
+import type Series from 'views/logic/aggregationbuilder/Series';
+import { parseSeries } from 'views/logic/aggregationbuilder/Series';
+
+const useWidgetUnits = (config: AggregationWidgetConfig) => {
+  const fieldTypesUnits = useFieldTypesUnits();
+  const usedFieldsInSeries = useMemo(() => config.series.map((s: Series) => {
+    const { field } = parseSeries(s.function) ?? {};
+
+    return field;
+  }), [config.series]);
+  const usedFieldsInRowPivots = useMemo(() => flatten(config.rowPivots.map((p) => p.fields)), [config.rowPivots]);
+  const usedFieldsInColumnPivots = useMemo(() => flatten(config.columnPivots.map((p) => p.fields)), [config.columnPivots]);
+
+  const allFields: Array<string> = useMemo(() => uniq(compact([...usedFieldsInSeries, ...usedFieldsInRowPivots, ...usedFieldsInColumnPivots])), [usedFieldsInColumnPivots, usedFieldsInRowPivots, usedFieldsInSeries]);
```

Comment:
We can do this with plain JS, without external dependencies:
```suggestion
  const allFields = useMemo(() => new Set([...usedFieldsInSeries, ...usedFieldsInRowPivots, ...usedFieldsInColumnPivots].filter(value => !!value)), [usedFieldsInColumnPivots, usedFieldsInRowPivots, usedFieldsInSeries]);
```

Reply:
I think with lodash functions it looks cleaner. Further in the code, we have filtration and need an array. So we would have to convert Set to an array. I'm not sure if removing dependencies worth so many manipulations.